front-end/my-vite-app/src/Survey.jsx

```diff
@@ -29,44 +33,84 @@ const Survey = ({closeModal}) => {
           })
           .then(data => {
             setCourses(data);
+            recommendations(data, level, rating, languages);
           })
           .catch(error => {
             console.error('Error fetching cards:', error);
           });
       };
 
-    const recommendations = (courses, level, rating, languages) => {
-        
-    }
+    const recommendations = async (courses, level, rating, languages) => {
+        const fetchAverageRating = async (courseId) => {
+            try {
+                const response = await fetch(`${import.meta.env.VITE_BACKEND_ADDRESS}/courses/${courseId}/average-rating`);
+                if (!response.ok) {
+                    throw new Error(`Failed to fetch average rating: ${response.status} ${response.statusText}`);
+                }
+                const data = await response.json();
+                return data.averageRating;
+            } catch (error) {
+                console.error('Error fetching average rating:', error);
+                return null;
+            }
+        };
+
+        const ratings = await Promise.all(courses.map(course => fetchAverageRating(course.title)));
```

Comment:
Can this be cached or stored somewhere? And you can update it every time the rating for a course is updated?

Reply:
Might be good to make this algorithm into it's own js module